filters/openpolicyagent/internal/envoy/skipperadapter.go

```diff
@@ -25,7 +29,7 @@ func AdaptToExtAuthRequest(req *http.Request, metadata *ext_authz_v3_core.Metada
 				Http: &ext_authz_v3.AttributeContext_HttpRequest{
 					Host:    req.Host,
 					Method:  req.Method,
-					Path:    req.URL.Path,
+					Path:    req.URL.RequestURI(),
```

Comment:
if you have filter chain:
```
setPath("/foo") -> opaAuthorizeRequest("...")
```

The request will be to http://.../my-path

Would you expect /my-path or /foo that is checked by the opa filter?

RequestURI will be /my-path

Reply:
Assuming that `req.URL.Path` would give /foo, the current behaviour would be that OPA filter/policy would see /foo as the path. So this change of using RequestURI would break that behaviour.

probably using below would be a way to retain the current behaviour + pass the query params
```
Path:    req.URL.Path + "?" + req.URL.RawQuery
```

Reply:
I don't know what is expected. From my side we can also document the behavior that req.URL.RequestURI() (link godoc stdlib) is used and would be fine for me.

Reply:
i felt if we use the `req.URL.RequestURI()` just to get the query params in the envoy check request, it will make the OPA filter not usable with setPath() and the behaviour will be a bit confusing for a user.

Decide to use a different approach to build the path with the query params.

Reply:
> Assuming that `req.URL.Path` would give /foo, the current behaviour would be that OPA filter/policy would see /foo as the path. So this change of using RequestURI would break that behaviour.
> 
> probably using below would be a way to retain the current behaviour + pass the query params
> 
> ```
> Path:    req.URL.Path + "?" + req.URL.RawQuery
> ```

Made a wrong assumption here 😢 
[req.URL.RequestURI()](https://github.com/golang/go/blob/af0c40311e2ee33ecd24971257606f42a49cf593/src/net/url/url.go#L1165-L1181) actually returns the path with the query params.

Therefore using req.URL.RequestURI() will not omit the path set by setPath() ([test case](https://github.com/zalando/skipper/pull/3207/files#diff-bea895a3252dbca936635114f887dfcdd5a66b7360d676ace6dab16e9a0d8099R77-R89)).

ie. 
setPath("/foo") -> opaAuthorizeRequest("...")
with request to http://.../my-path, the OPA filter would pass "foo" as the path.